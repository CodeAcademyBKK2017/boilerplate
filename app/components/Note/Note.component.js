import Overlay from 'react-native-modal-overlay';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Note.style';
import Touchable from 'react-native-platform-touchable';
import {FlatList, Text, View} from 'react-native';

export default class Note extends Component {
    state ={
      modalVisible: false,
      title: '',
      content: ''
    }
 
onOpen=(title, content) => () => {
  this.setState({
    modalVisible: !this.state.modalVisible
    , title
    , content});
}
onClose=() => {
  this.setState({modalVisible: false});
}
generateList = ({item}) => (
  <Touchable style={styles.box} onPress={this.onOpen(item.title, item.content)}>
    <View > 
      <Text style={styles.title}>{item.title}</Text>
      <Text style={styles.content}>{item.content}</Text>
    </View>
  </Touchable>)
render () {
       
  return (
    <View style={styles.container}>
      <Text style={styles.header} >Notes:</Text>
      <FlatList data={this.props.noteList} renderItem={this.generateList} />
      <Overlay visible={this.state.modalVisible}
        closeOnTouchOutside={true}  
        onClose={this.onClose}
        animationType='slideInLeft'
        containerStyle={styles.modalContainer}
        childrenWrapperStyle={styles.modalChildrenWrap}
        animationDuration={1000}>
        <Text style={styles.title}> {this.state.title}</Text>
        <Text style={styles.content}> {this.state.content}</Text>
      </Overlay>
    </View>
  );
}
}

Note.propTypes = {
  noteList: PropTypes.array
};
  