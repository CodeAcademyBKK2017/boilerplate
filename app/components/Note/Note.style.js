import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  title: {
    fontWeight: 'bold',
    fontSize: 15
  },
  content: {
    padding: 10,
    fontSize: 15
  },
  header: {
    fontWeight: 'bold'
  },
  container: {
    flex: 1,
    width: '100%'
  },
  box: {
    paddingVertical: 10
  },
  modalContainer: {
    backgroundColor: 'rgba(37, 8, 10, 0.78)'
  },
  modalChildrenWrap: {
    backgroundColor: '#eee',
    alignItems: 'flex-start'
  },
  noteBlock: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  del: {
    color: 'red'
  }
});