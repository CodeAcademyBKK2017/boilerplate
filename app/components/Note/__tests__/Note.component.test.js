import 'react-native';
import noop from 'lodash/noop';
import Note from '../Note.component';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('Note', () => {
  let wrapper, instance;
  beforeEach(() => {
    wrapper = shallow(<Note onDelete={noop} />);
    instance = wrapper.instance();
  });
  it('snapshot test', () => {
    const snapshot = renderer.create(<Note />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('generateList should have been render some JSX', () => {
    const item =  {title: 'someTitle', content: 'someContent', key: 'someUUID'};
    const snapshot = instance.generateList({item});
    expect(snapshot).toMatchSnapshot();
  });
  it('onOpen should have state change', () => {
    const title = 'xx';
    const content = 'yy';
    const expectedState = {
      modalVisible: true,
      title: 'xx',
      content: 'yy'
    };
    instance.onOpen(title, content)();
    expect(instance.state).toEqual(expectedState);
  });
  it('onClose should have state change', () => {
    const expectedInitialstate = {
      modalVisible: false,
      title: '',
      content: ''
    };
    instance.onClose();
    expect(instance.state).toEqual(expectedInitialstate);
  });
});