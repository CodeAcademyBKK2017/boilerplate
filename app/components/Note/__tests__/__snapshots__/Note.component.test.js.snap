// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note generateList should have been render some JSX 1`] = `
<PlatformTouchable
  onPress={[Function]}
  style={
    Object {
      "paddingVertical": 10,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      someTitle
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "padding": 10,
        }
      }
    >
      someContent
    </Text>
  </View>
</PlatformTouchable>
`;

exports[`Note snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 0.3,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Notes:
  </Text>
  <RCTScrollView
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <Modal
    animationOutType="fadeOut"
    animationType="none"
    hardwareAccelerated={false}
    onClose={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(37, 8, 10, 0.78)",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0,
          "padding": 20,
        }
      }
      testID={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "padding": 20,
            },
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#eee",
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
        >
           
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "padding": 10,
            }
          }
        >
           
          
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
